client/lib/domains/index.js

```diff
@@ -58,9 +58,11 @@ function checkDomainAvailability( domainName, onComplete ) {
 			if ( unmappabilityReason ) {
 				errorCode += `_${ unmappabilityReason }`;
 			}
-		} else if ( ! isAvailable && isMappable ) {
+		} else if ( isAvailable === null && isMappable ) {
+			errorCode = 'no_availability_but_mappable';
+		} else if ( isAvailable === false && isMappable ) {
```

Comment:
In general, I agree - but don't you think this is a valid use case? It's nicely backwards compatible, while offering an option to differentiate between an unavailable domain and a one we can't check availability for. An alternative would be to change `is_available` to have a string value... but that would require a v2 of the endpoint.
I understand that explicit checks for `false` and `null` seem like a code smell - but that's actually what I like about this change. If all you need is a check if it's available, you can do us as normal: isAvailable / ! isAvailable. But if you need something more fine-grained, you can differentiate using `false` and `null`. `null` seems completely at place here - it shows it's an empty value.